### DISCUSSION TITLE
API filter of relational data generates invalid SQL

### DISCUSSION THREAD

**Discussion by tenebrius** [directus/directus#16022] [Archive]
### Describe the Bug

Querying deeply nested relational data generates an invalid SQL using the api filter

### To Reproduce

Have a data structure like this

One `TEAM `has many `PLAYERS`
One `TEAM  `has many `AWARDS`

Using API, query the `PLAYERS` collection using this filter
```
team: {
  awards: {
    type: "regional",
    year: "2022"
  }
}
```

You will get the following error
```
select `player`.`id`, `player`.`player_name`, `player`.`team`, `player`.`rating`, `player`.`age` from `player` left join `team` as `okqtf` on `player`.`team` = `okqtf`.`id` left join `award` as `nnzmq` on `okqtf`.`id` = `nnzmq`.`team` where (`okqtf`.`awards` = {"type":{"_eq":"regional"}}, {"year":{"_eq":"2022"}}) order by `player`.`id` asc limit 100 - SQLITE_ERROR: no such column: okqtf.awards
```

## SQLite database file: 
file: [data.zip](https://github.com/directus/directus/files/9794105/data.zip)

with admin login
```
login: admin@example.com
password: 123456
```

## Codesanbox

Backend
`https://codesandbox.io/s/musing-morning-s8rtuk`

API Client
`https://codesandbox.io/s/fetch-api-example-forked-y4cxet?file=/src/index.js:164-252`

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

18

### What database are you using?

SQLite 5.12.0

### What browser are you using?

Chrome

### How are you deploying Directus?

locally

**Comment by licitdev** [comment]
The filter looks to be malformed. Kindly refer to https://docs.directus.io/reference/filter-rules.html for more information.

Here's an example filter rule which requires the player's team's to have an award that has the type equal to `regional` and year equal to `2022`.
```json
{
  "_and": [
    { "team": { "awards": { "type": { "_eq": "regional" } } } },
    { "team": { "awards": { "year": { "_eq": "2022" } } } }
  ]
}
```

**Reply by tenebrius** [reply]
Thanks. That did solve the issue.
